index.js

```diff
@@ -87,7 +87,9 @@ module.exports = function (config) {
 
   const log = function () {
     if (!config.quiet) {
-      if (config.logToStdErr) {
+      if (config.hasOwnProperty('logTransporter') && typeof config.logTransporter === 'function') {
+        config.logTransporter(arguments);
+      } else if (config.logToStdErr) {
```

Comment:
`logTransporter` is a little clunky in its naming. I prefer `logger`. Also I think it's better if the function resembles `console.log` in taking multiple arguments rather than an array of arguments (and currently all logging is just a single argument anyway). It's fine to just check whether `config.logger` exists for now. From a user perspective, if it isn't a function, it's better for it to throw an error than to be skipped entirely (though ideally it'd throw a specific error clearly saying it should be a function).

```suggestion
      if (config.logger) {
        config.logger(...arguments);
      } else if (config.logToStdErr) {
```